### PR TITLE
Packaging: Port fix for GetParts from .NETFramework

### PR DESCRIPTION
**Pull request by ericstj**
This ports a fix for the `GetParts` method to improve part URI handling and collision detection.  Fix in .NETFramework originally made by @Kuldeep-MS 

`src/libraries/System.IO.Packaging/src/System/IO/Packaging/Package.cs`: Added sorting of parts array to ensure proper order, introduced a dictionary and list to track parts and detect collisions, and implemented a new method `CopyPartDictionaryToPartList` to copy parts to `_partList`.

### CODE REVIEW

File: src/libraries/System.IO.Packaging/src/System/IO/Packaging/Package.cs [hunk 1]
```diff
@@ -401,32 +401,63 @@ public PackagePartCollection GetParts()
...
+                Array.Sort(parts, Comparer<PackagePart>.Create((partA, partB) => uriComparer.Compare((PackUriHelper.ValidatedPartUri)partA.Uri, (PackUriHelper.ValidatedPartUri)partB.Uri)));
+
                 //We need this dictionary to detect any collisions that might be present in the
                 //list of parts that was given to us from the underlying physical layer, as more than one
                 //partnames can be mapped to the same normalized part.
                 //Note: We cannot use the _partList member variable, as that gets updated incrementally and so its
                 //not possible to find the collisions using that list.
                 //PackUriHelper.ValidatedPartUri implements the IComparable interface.
-                Dictionary<PackUriHelper.ValidatedPartUri, PackagePart> seenPartUris = new Dictionary<PackUriHelper.ValidatedPartUri, PackagePart>(parts.Length);
+                Dictionary<string, KeyValuePair<PackUriHelper.ValidatedPartUri, PackagePart>> partDictionary = new Dictionary<string, KeyValuePair<PackUriHelper.ValidatedPartUri, PackagePart>>(parts.Length);
```
Comment (steveharter) [inline src/libraries/System.IO.Packaging/src/System/IO/Packaging/Package.cs:415]: This is a longer version of https://github.com/dotnet/runtime/pull/108677/commits/b4ccde403bdb8976d3ad4768d16002fa232f1ac0#diff-e0565b90c6887f4c5e77f09ebf8ae412b72414d0d4a9c49fe3bdf7b82525b81cR417

Expected?

File: src/libraries/System.IO.Packaging/src/System/IO/Packaging/Package.cs [hunk 2]
```diff
@@ -401,32 +401,63 @@ public PackagePartCollection GetParts()
 
                 PackUriHelper.ValidatedPartUri partUri;
 
+                var uriComparer = Comparer<PackUriHelper.ValidatedPartUri>.Default;
```
Comment (steveharter) [inline src/libraries/System.IO.Packaging/src/System/IO/Packaging/Package.cs:404]: Note above on LOC 393 there should be 2 more doc'd exceptions (FileFormat and InvalidOperation) based on https://github.com/dotnet/runtime/pull/108677/commits/b4ccde403bdb8976d3ad4768d16002fa232f1ac0#diff-e0565b90c6887f4c5e77f09ebf8ae412b72414d0d4a9c49fe3bdf7b82525b81cR383-R384

**Review by steveharter — APPROVED** [review]
LGTM; some minor questions on syntax and doc

**Comment by ericstj** [comment]
/ba-g unrelated nuget audit warnings already fixed in main

**Pull request merged**